lib/src/ast/element.dart

```diff
@@ -134,24 +169,30 @@ abstract class ElementAst implements StandaloneTemplateAst {
   }
 }
 
-class _ParsedElementAst extends TemplateAst with ElementAst {
-  final NgToken _nameToken;
-
-  _ParsedElementAst(
-    SourceFile sourceFile,
-    NgToken beginToken,
-    this._nameToken,
-    NgToken endToken, {
-    this.attributes: const [],
-    this.childNodes: const [],
-    this.events: const [],
-    this.properties: const [],
-    this.references: const [],
-  })
+class ParsedElementAst extends TemplateAst with ElementAst {
+  final NgToken identifierToken;
+  final int openTagEnd;
+  final int closeTagStart; //Can be null if void element
```

Comment:
Comment like that should be public, i.e.

/// May be null if a void `<tag />` element (no closing tag).